test/jdk/javax/imageio/ImageReaderReadAll.java

```diff
@@ -87,8 +88,7 @@ public BufferedImage read(int imageIndex, ImageReadParam param)
           throws IOException {
             if (input == null)
                 throw new IllegalStateException();
-            if (imageIndex >= 1 || imageIndex < 0)
-                throw new IndexOutOfBoundsException();
+            Objects.checkIndex(imageIndex, 1);
             if (seekForwardOnly) {
                 if (imageIndex < minIndex)
                     throw new IndexOutOfBoundsException();
```

Comment:
Can't we use checkFromToIndex​ for this IOOBE too? Used in many other places in this PR...

Please also confirm all tests modified are green after this modification..

Reply:
checkFromToIndex​ has different semantice with checkIndex:

- checkFromToIndex
```
     * <ul>
     *  <li>{@code fromIndex < 0}</li>
     *  <li>{@code fromIndex > toIndex}</li>
     *  <li>{@code toIndex > length}</li>
     *  <li>{@code length < 0}, which is implied from the former inequalities</li>
     * </ul>
```
- checkIndex
```
     * <ul>
     *  <li>{@code index < 0}</li>
     *  <li>{@code index >= length}</li>
     *  <li>{@code length < 0}, which is implied from the former inequalities</li>
     * </ul>
```
---

The following tests are passed after applying this patch:
```
test/jdk/javax/imageio/AppletResourceTest.java
test/jdk/javax/imageio/ImageReaderReadAll.java
test/jdk/javax/imageio/metadata/IIOMetadataFormat/UserPluginMetadataFormatTest.java
```

ClippedImages.java is a manual test, I have no environment to test it, but since it's a trivial replacement, I think it's okay..

Reply:
I meant 
```
if (imageIndex < minIndex)
                    throw new IndexOutOfBoundsException();
```
can't it be replaced with Objects.checkFromToIndex(minIndex, imageIndex, ..

Reply:
Hi, sorry for the late response. checkFromToIndex has many constraints, this code may not satisfy those constraints, so I don't think we should do that in this PR.(Maybe we can file a new PR to do replacement carefully)

Reply:
OK...The subject line of this PR states that checkFromToIndex will also be used but I see only checkIndex so I intervened. Anyways, the code in question is in test so not much of an issue.

Reply:
Thanks @prsadhuk for review! I've updated the title to exclude checkFromToIndex.